bridges/BookMyShowBridge.php

```diff
@@ -0,0 +1,1389 @@
+	}
+
+	private function generateMoviesData($eventGroup){
+		// Additional data picked up from the first Child Event
+		$data = $eventGroup['ChildEvents'][0];
+
+		return array(
+			'uri' => $this->generateMovieUrl($eventGroup),
+			'title' => $eventGroup['EventTitle'],
+			'timestamp' => $data['EventDate'],
+			'author' => 'BookMyShow',
+			'content' => $this->generateMovieHtml($eventGroup),
+			'enclosures' => array(
+				sprintf(self::MOVIES_IMAGE_BASE_FORMAT, $data['EventImageCode']),
+			),
+			// Sample Input = |ADVENTURE|ANIMATION|COMEDY|
+			// Sample Output = ['Adventure', 'Animation', 'Comedy']
+			'categories' => array_filter(
+				explode('|', ucwords(strtolower($eventGroup['EventGrpGenre']), '|'))
+			),
+			'uid' => $eventGroup['EventGroup']
+		);
+	}
+
+	private function generateEventData($event, $category){
+		if($category == self::MOVIES) {
+			return $this->generateMoviesData($event);
+		}
+
+		return $this->generateGenericEventData($event, $category);
+	}
+
+	/**
+	 * Takes an event data as array and returns data for RSS Post
+	 */
+	private function generateGenericEventData($event, $category){
+		return array(
+			'uri' => $event['FShareURL'],
+			'title' => $event['EventTitle'],
+			'timestamp' => $event['Event_dtmCreated'],
+			'author' => 'BookMyShow',
+			'content' => $this->generateEventHtml($event, $category),
+			'enclosures' => array(
+				$event['BannerURL'],
+			),
+			'categories' => array_merge(
+				array(self::CATEGORIES[$category]),
+				$event['GenreArray']
+			),
+			'uid' => $event['EventGroupCode']
+		);
+	}
+
+	public function collectData(){
+		$city = $this->getInput('city');
+		$category = $this->getInput('category');
+
+		$url = $this->makeUrl($category);
+		$headers = $this->makeHeaders($city);
+
+		$data = json_decode(getContents($url, $headers), true);
+
+		if ($category == self::MOVIES) {
+			$data = $data['moviesData']['BookMyShow']['arrEvents'];
+		} else {
+			$data = $data['data']['BookMyShow']['arrEvent'];
+		}
+
+		foreach ($data as $event) {
+			$item = $this->generateEventData($event, $category);
+			if ($this->matchesFilters($category)) {
+				$this->items[] = $item;
+			}
+		}
+	}
+
+	/**
+	 * Currently only checks if the language filter matches
+	 */
+	private function matchesFilters($category){
+		if ($this->getInput('language') !== 'all') {
+			$language = $this->getInput('language');
+			return in_array($language, $this->languages);
+		}
+		return true;
+	}
+
+	/**
+	 * Generates the RSS Feed title
+	 */
+	public function getName(){
+		$city = $this->getInput('city');
+		$category = $this->getInput('category');
+		if(!is_null($city) and !is_null($category)) {
+			$categoryName = self::CATEGORIES[$category];
+			$cityNames = array_flip(self::CITIES);
+			$cityName = $cityNames[$city];
+			if ($this->getInput('language')!=='null') {
+				$l = $this->getInput('language');
+				// Sample: English Movies in Delhi
+				return "BookMyShow: $language $categoryName in $cityName";
```

Comment:
I get an error at this line if language is set to 'all':

```
Undefined variable: language in /var/www/html/rss-bridge/bridges/BookMyShowBridge.php on line 1341
```

Reply:
I suppose line 1338 is wrong. Shouldn't it be `if ($this->getInput('language') !== 'all') {`?